contracts/protocol/v1/proxies/P1SoloBridgeProxy.sol

```diff
@@ -255,20 +285,28 @@ contract P1SoloBridgeProxy is
         if (msg.sender != transfer.account) {
             I_Solo solo = I_Solo(_SOLO_MARGIN_);
             I_PerpetualV1 perpetual = I_PerpetualV1(transfer.perpetual);
+            bytes32 transferMode = _getTransferMode(transfer);
+            bool toPerpetual = (
+                transferMode == TRANSFER_MODE_SOME_TO_PERPETUAL ||
+                transferMode == TRANSFER_MODE_ALL_TO_PERPETUAL
+            );
```

Comment:
nit: the logic of these lines can be encapsulated in `_hasWithdrawPermissions()`